### DISCUSSION TITLE
Is data exchange with data spaces using other connector implementations possible?

### DISCUSSION THREAD

**Discussion by epanagou-ubitech** [eclipse-edc/Connector#3527] [Q&A] [answered]
The scenario we have in mind:

We have set up **Data Space A** using the **EDC Connector** and there is also an existing **Data Space X** using **Connector implementation X** with which we would like to make a data exchange.

For the data exchange we are thinking of following the steps below:

1. Join Data Space X and perform a contract agreement in order to be able to consume the asset we are interested in, e.g. **Asset X**.
2. We create an asset in Data Space A, **Asset A** using the associated transfer endpoint of Asset X as the Data Source endpoint.
3. We perform a contract agreement for Asset A and then perform a transfer to receive the asset data.

Our questions are:
- a) Would these steps be correct?
- b) Is there a simpler method of performing this data exchange?
- c) And slightly off-topic: Is there compatibility between connector implementations so that a data space may be formed using different connector implementations?

Thank you in advance!

**Comment by jimmarino** [comment]
I'm not sure I understand the question(s) being asked:

1. EDC will conform to the [DSP specifications](https://github.com/International-Data-Spaces-Association/ids-specification) once the conformance test kit is completed and it passes the tests. There is no official timeframe for this. 
2. EDC can only make a statement about conformance, not whether it is "compatible" with another connector implementation, as use cases and dataspace configuration may vary.
3. I don't understand points a-b. Please review the DSP specifications, particularly the [Dataspace Model](https://github.com/International-Data-Spaces-Association/ids-specification/blob/main/model/model.md). Each EDC instance by design has exactly one identity. If a participant has a stable identity across dataspaces, a single instance may transact in those dataspaces. That is an unlikely scenario.

I don't understand what "data source endpoint" is, so it is difficult to comment further.

**Reply by epanagou-ubitech** [reply]
Thank you for getting back to us!

If we consider the following diagram of a data space in the [current DSP documentation](https://github.com/International-Data-Spaces-Association/ids-specification/blob/7d468461d202698e2fac819b855b36f6ca90194e/README.md) is there a requirement that the connector implementation used for the connector on the left and the connector on the right are the same? We assume that both implementations conform to the DSP.

![image](https://github.com/eclipse-edc/Connector/assets/133344516/b5479480-44cb-4b5f-b66a-1d156d1fe9e3)

- By "data source endpoint" we mean either a database and/or a REST endpoint that provides the data associated with an asset, as seen in the previous diagram.
- Essentially, our main question is, is there method for two separate dataspaces to exchange data?

**Reply by jimmarino** [reply, marked as answer]
OK, that is a high-level, non-normative document that is not part of the specification. The conceptual model is defined here:

https://github.com/International-Data-Spaces-Association/ids-specification/blob/7d468461d202698e2fac819b855b36f6ca90194e/model/model.md

There is no requirement *at the specification level* for two connectors to be the same implementation. However, in practice that may be the case if a dataspace makes use of proprietary features.

**Reply by epanagou-ubitech** [reply]
Thank you!
